Arrays PR Request (#2920)
Codecademy/docs: content/lua/concepts/arrays/arrays.md

Description: 'Arrays are used to store ordered collections of data.' # Required; ideally under 150 characters and starts with a noun (used in search engine results and content previews)
Subjects: # Please only use Subjects in the subjects.md file (https://github.com/Codecademy/docs/blob/main/documentation/subjects.md). If that list feels insufficient, feel free to create a new Subject and add it to subjects.md in your PR!
  - 'Computer Science'
Tags: # Please only use Tags in the tags.md file (https://github.com/Codecademy/docs/blob/main/documentation/tags.md). If that list feels insufficient, feel free to create a new Tag and add it to tags.md in your PR!
  - 'Tables'
  - 'Arrays'
  - 'Iterators'
CatalogContent: # Please use course/path landing page slugs, rather than linking to individual content items. If listing multiple items, please put the most relevant one first
  - 'learn-lua'
  - 'paths/computer-science'
---

# Arrays

**Arrays** are tables of ordered, stored **data**. They can hold any type of data, and can be cleared with `nil` values.

## Syntax

> In Lua, Arrays can be created with curly braces `{}`. When storing values, arrays will automatically order values starting from `0` unless specified otherwise. For instance, the following code will store 100 values using an `iteration`.

Reviewer: From what I've seen, the default in Lua, is to begin with 1, in contrast to most languages. For example:
foo = {10, 20, 30}
print(foo[1]) -- yields 10

```lua

a = {};

for i = 1, 100 do
    a[i] = 0
end

```

When the array is printed, it would display 100 values of 0 starting at an index of 1. However, if we try printing any value out of this range, our output will return **nil** instead. This is because **nil** refers to data that is not there, which in our case would be true because our example only stores 100 values.

## Accessing Elements within the Array

To access an array element, we reference the desired value by its **index**. For example, to reference a value from the following array:

```lua
a = {
    [1] = “🎁”, -- "Gift"
    [2] = “🍌”, -- "Banana"
    [3] = “🍎” -- "Apple"
}
```

> Here, there is an array of elements that range from the indexes 1 - 3. To print “🍌”, the value must be referenced through its assigned index, which in this case is 2.

Reviewer: I think this section should probably be expanded to include examples that demonstrate overwriting or assignment of new values and (as mentioned above) clearing or removing values with nil.

```lua
a = {
    [1] = “🎁”, -- "Gift"
    [2] = “🍌”, -- "Banana"
    [3] = “🍎” -- "Apple"
}

print(a[2]) -- “🍌” [Banana]
```

> This can be done for every value within an array. This also works with **dictionaries**, but instead the assigned **key** must be used to retrieve a value. A key is essentially an index with the only difference being that a key may be assigned as any **datatype**, while an index can only come in the form of a number.

Reviewer: ```suggestion
This can be done for every value within an array. This also works with dictionaries, but instead the assigned key must be used to retrieve a value. A key is essentially an index with the only difference being that a key may be assigned as any **datatype**, while an index can only come in the form of a number.
```

For example:

```lua
dictionary = {
    [“🍎”] = “Apple”,
    [“🍌”] = “Banana”,
    [“🍊”] = “Orange”
}

print(dictionary[“🍊”]) -- “Orange”
```
  

> It is important to note that Lua will not print the key of the dictionary being used, but will instead print the value related to the requested key. This behavior also applies to **Arrays**.

Reviewer: ```suggestion
It is important to note that Lua will not print the key of the dictionary being used, but will instead print the value related to the requested key. This behavior also applies to arrays.
```

Subjects: # Please only use Subjects in the subjects.md file (https://github.com/Codecademy/docs/blob/main/documentation/subjects.md). If that list feels insufficient, feel free to create a new Subject and add it to subjects.md in your PR!
  - 'Computer Science'
  - 'Lua'
Tags: # Please only use Tags in the tags.md file (https://github.com/Codecademy/docs/blob/main/documentation/tags.md). If that list feels insufficient, feel free to create a new Tag and add it to tags.md in your PR!
  - 'Tables'
  - 'Arrays'
  - 'Iterators'
CatalogContent: # Please use course/path landing page slugs, rather than linking to individual content items. If listing multiple items, please put the most relevant one first
  - 'lua/arrays'
---

# Arrays

**Arrays** are tables of ordered, stored **data**. They can hold any type of data, and can be cleared with `nil` values.

## Syntax

In Lua, Arrays can be created with curly braces `{}`. When storing values, arrays will automatically order values starting from `0` unless specified otherwise. For instance, the following code will store 100 values using an `iteration`.

```lua

a = {};

for i = 1, 100 do
    a[i] = 0
end

```

> If we print our array, we see that it now holds 100 values starting from 1. However, if we try printing any value out of this range, we will receive **nil** instead. This is because **nil** refers to data that is not there, which in our case would be true because we have only stored 100 values.

Reviewer 2: I also think we can try to minimize the amount of "we" and "you" we use in these entries. These are more similar to our instructional content in courses.

> If we print our array, we see that it now holds 100 values starting from 1. However, if we try printing any value out of this range, we will receive **nil** instead. This is because **nil** refers to data that is not there, which in our case would be true because we have only stored 100 values.

Reviewer 2: For example, 
`If we print our array, we see that it now holds 100 values starting from 1`
can become
`When the array is printed, it would display 100 values starting from 1`

> If we print our array, we see that it now holds 100 values starting from 1. However, if we try printing any value out of this range, we will receive **nil** instead. This is because **nil** refers to data that is not there, which in our case would be true because we have only stored 100 values.

Author: What I meant to say was that our index starts at 1. Also, I'll apply these changes to minimize the usage of "we" and "you".

## Accessing Elements within the Array

When working with **Array’s**, we may want to reference an element within it. To do this, we reference the desired value by its **index**.
For example, let's say we want to reference a value from the following array:

```lua
a = {
    [1] = “🎁”, -- "Gift"
    [2] = “🍌”, -- "Banana"
    [3] = “🍎” -- "Apple"
}
```

> Here, we are provided with an array of elements that range from the indexes 1 - 3. Let's say we want to print “🍌”. To achieve this, we must reference our value through its assigned index, which in our case is 2.

Reviewer 2: ```suggestion
Here, there is an array of elements that range from the indexes 1 - 3. To print “🍌”, the value must be referenced through its assigned index, which in this case is 2.
```

```lua
a = {
    [1] = “🎁”, -- "Gift"
    [2] = “🍌”, -- "Banana"
    [3] = “🍎” -- "Apple"
}

print(a[2]) -- “🍌” [Banana]
```

> We can do this for every value within our array. This also works with **dictionaries**, but instead we must use the assigned **key** to retrieve a value. A key is essentially an index with the only difference being that a key may be assigned as any **datatype**, while an index can only come in the form of a number.

Reviewer 2: ```suggestion
This can be done for every value within an array. This also works with **dictionaries**, but instead the assigned **key** must be used to retrieve a value. A key is essentially an index with the only difference being that a key may be assigned as any **datatype**, while an index can only come in the form of a number.
```

For example:

```lua
dictionary = {
    [“🍎”] = “Apple”,
    [“🍌”] = “Banana”,
    [“🍊”] = “Orange”
}

print(dictionary[“🍊”]) -- “Orange”
```
  

> It is important to note that Lua will not print the key of the dictionary we are using, but will instead print the value related to the requested key. This behavior also applies to **Arrays**.

Reviewer 2: ```suggestion
It is important to note that Lua will not print the key of the dictionary being used, but will instead print the value related to the requested key. This behavior also applies to **Arrays**.
```
